optuna_dashboard/_app.py

```diff
@@ -449,6 +451,60 @@ def save_trial_note(study_id: int, trial_id: int) -> dict[str, Any]:
         response.status = 204  # No content
         return {}
 
+    @app.get("/csv/<study_id:int>")
+    def download_csv(study_id: int) -> BottleViewReturn:
+        # Create a CSV file
+        summary = get_study_summary(storage, study_id)
+        if summary is None:
+            response.status = 404  # Not found
+            return {"reason": f"study_id={study_id} is not found"}
+        trials = get_trials(storage, study_id)
+
+        param_names = []
+        user_attr_names = []
+        for trial in trials:
+            for param_name in trial.params.keys():
+                if param_name not in param_names:
+                    param_names.append(param_name)
+            for attr_name in trial.user_attrs.keys():
+                if attr_name not in user_attr_names:
+                    user_attr_names.append(attr_name)
```

Comment:
Is it more concise?

```suggestion
        from itertools import chain

        param_names = sorted(set(chain.from_iterable([t.params.keys() for t in trials])))
        user_attr_names = sorted(set(chain.from_iterable([t.user_attrs.keys() for t in trials])))
```

Reply:
I reconsidered and conciseness is more important than the order. Thank you for your comment.